spec/hydra/works/models/concerns/generic_file/characterization_spec.rb

```diff
@@ -0,0 +1,216 @@
+        end
+
+        it "has correct values in array for properties that require single values" do
+          expect(subject.filename).to eq("sample-file.pdf")
+          expect(subject.mime_type).to eq("application/pdf")
+        end
+      end
+    end
+  end # default parse_config context
+
+  context "using image module" do
+    let(:characterization)     { class_double("Hydra::FileCharacterization").as_stubbed_const }
+    let(:fits_filename)        { 'fits_0.8.5_jp2.xml' }
+    let(:fits_response)        { IO.read(File.join(fixture_path, fits_filename)) }
+    let(:file)                 { Hydra::PCDM::File.new }
+    let(:image_class) do
+      Class.new(Hydra::Works::GenericFile::Base) do
+        include Hydra::Works::GenericFile::Characterization
+        include Hydra::Works::GenericFile::Characterization::Image
+      end
+    end
+
+    subject { image_class.new(id: 'demo/02/02') }
+
+    before do
+      mock_add_file_to_generic_file(subject, file)
+      allow(characterization).to receive(:characterize).and_return(fits_response)
+    end
+
+    it "has expected values for added properties" do
+      subject.characterize
+
+      expect(subject.file_size).to eq(["11043"])
+      expect(subject.byte_order).to eq("big endian")
+      expect(subject.compression).to eq("JPEG 2000")
+      expect(subject.width).to eq("512")
+      expect(subject.height).to eq("465")
+      expect(subject.color_space).to eq("sRGB")
+    end
+  end
+
+  context "using video module" do
+    let(:characterization)     { class_double("Hydra::FileCharacterization").as_stubbed_const }
+    let(:fits_filename)        { 'fits_0.8.5_avi.xml' }
+    let(:fits_response)        { IO.read(File.join(fixture_path, fits_filename)) }
+    let(:file)                 { Hydra::PCDM::File.new }
+    let(:video_class) do
+      Class.new(Hydra::Works::GenericFile::Base) do
+        include Hydra::Works::GenericFile::Characterization
+        include Hydra::Works::GenericFile::Characterization::Video
+      end
+    end
+
+    subject { video_class.new(id: 'demo/03/03') }
+
+    before do
+      mock_add_file_to_generic_file(subject, file)
+      allow(characterization).to receive(:characterize).and_return(fits_response)
+    end
+
+    it "has expected values for added properties" do
+      subject.characterize
+
+      expect(subject.height).to eq(["264"])
+      expect(subject.width).to eq(["356"])
+      expect(subject.duration).to eq(["14.10 s"])
+      expect(subject.frame_rate).to eq(["10"])
+      expect(subject.sample_rate).to eq(["11025"])
+    end
+  end
+
+  context "using audio module" do
+    let(:characterization)     { class_double("Hydra::FileCharacterization").as_stubbed_const }
+    let(:fits_filename)        { 'fits_0.8.5_mp3.xml' }
+    let(:fits_response)        { IO.read(File.join(fixture_path, fits_filename)) }
+    let(:file)                 { Hydra::PCDM::File.new }
+    let(:video_class) do
+      Class.new(Hydra::Works::GenericFile::Base) do
```

Comment:
An application can still have an everything bagel.  If the repo can expect all of the types, as curation concerns does, including all four media type modules will be the course of action.

I went with the a la cart model because the application may have more apriori information about an uploaded file or the nature of the repository.  For example, HathiTrust doesn't take Audio or Video so having those properties would be clutter at best.

Or it may be known that only the simplest characterization will be useful because the files that you're going to be ingesting may have been emitted by some proprietary scientific appliance or software package.  Those files can be pathological in that they'll appear to be a valid file type but interact badly or unexpectedly with other software.